src/components/map/actions/measure-action.js

```diff
@@ -1,77 +1,160 @@
 import Overlay from 'ol/Overlay';
 import { unByKey } from 'ol/Observable';
+import GeometryType from 'ol/geom/GeometryType';
 import { VlDrawAction } from './draw-action';
+import '../../pill';
 
 export class VlMeasureAction extends VlDrawAction {
   constructor(layer, options) {
     super(
       layer,
-      'LineString',
+      GeometryType.LINE_STRING,
       () => {
         unByKey(this.measurePointermoveHandler);
       },
-      options,
+      { ...options, maxPoints: 2 },
     );
 
-    let featureCounter = 0;
-
+    this.featureCounter = 0;
     this.layer = layer;
-    this.measureTooltips = [];
+    this.measurementTooltips = [];
     this.measurePointermoveHandler = undefined;
 
-    const showMeasureTooltip = (feature, tooltip, tooltipElement) => {
-      const length = feature.getGeometry().getLength().toFixed(2);
-      tooltipElement.textContent = `${length} m`;
-      tooltip.setElement(tooltipElement);
-      tooltip.setPosition(feature.getGeometry().getLastCoordinate());
-    };
-
-    this.drawInteraction.on('drawstart', (event) => {
-      const id = featureCounter++;
-      const measureFeature = event.feature;
-      measureFeature.setId(id);
-      const tooltipElement = document.createElement('div');
-      tooltipElement.setAttribute('class', 'measure-tooltip');
-      const tooltip = new Overlay({
-        offset: [-15, 10],
-        positioning: 'bottom-center',
-      });
-      this.map.addOverlay(tooltip);
-      this.measureTooltips[id] = tooltip;
-      this.measurePointermoveHandler = this.map.on('pointermove', () =>
-        showMeasureTooltip(measureFeature, tooltip, tooltipElement),
-      );
+    this.measureOptions = options;
+  }
+
+  activate() {
+    this.drawStartHandler = this.drawInteraction.on('drawstart', (event) => {
+      this._handleDrawStart(event);
+    });
+
+    this.removeFeatureHandler = this.layer.getSource().on('removefeature', (event) => {
+      this._handleRemoveFeature(event);
+    });
+
+    super.activate(this);
+  }
+
+  _setMeasurementTooltipsClosable(closable) {
+    this.measurementTooltips.forEach((tooltip) => {
+      // Check if tooltip still exists
+      if (tooltip) {
+        if (tooltip.getElement()) {
+          if (closable) {
+            tooltip.getElement().setAttribute('data-vl-closable', closable);
+          } else {
+            tooltip.getElement().removeAttribute('data-vl-closable');
+          }
+        }
+      }
+    });
+  }
+
+  _showMeasurementTooltip(feature, tooltipOverlay, tooltipElement) {
+    const length = feature.getGeometry().getLength().toFixed(2);
+    tooltipElement.textContent = `${length} m`;
+    tooltipOverlay.setElement(tooltipElement);
+    tooltipOverlay.setPosition(feature.getGeometry().getLastCoordinate());
+  }
+
+  _handleDrawStart(event) {
+    // Add measurement line (feature) and tooltip (overlay)
```

Comment:
Het is de bedoeling dat als zo'n kaartlaag kan worden aan/uit getoggled zoals besproken met de `vl-map-layer-switcher`. Als ik eens zo'n demo uitwerk, merk ik dat ik de metingen wel togglebaar zijn, maar de pills blijven staan.

Reply:
![Schermafbeelding 2022-04-28 om 16 16 27](https://user-images.githubusercontent.com/22538537/165773149-e6eba470-18e1-47a6-b647-897217acd26d.png)

Reply:
fixed